app/models/concerns/foreman_remote_execution/smart_proxy_extensions.rb

```diff
@@ -19,13 +23,24 @@ def update_pubkey
       key
     end
 
+    def update_ca_pubkey
+      return unless has_feature?(%w(SSH Script))
+
+      key = ::ProxyAPI::RemoteExecutionSSH.new(:url => url).ca_pubkey
+      # smart proxy is not required to have a CA pubkey, in which case an empty string is returned
+      key = nil if key == ""
```

Comment:
This reads a bit nicer and should do the same thing
```suggestion
      # smart proxy is not required to have a CA pubkey, in which case an empty string is returned
      key = ::ProxyAPI::RemoteExecutionSSH.new(:url => url).ca_pubkey&.presence
```